tests/FunctionalTestCase.php

```diff
@@ -497,6 +502,33 @@ protected function skipIfClientSideEncryptionIsNotSupported(): void
         }
     }
 
+    protected function skipIfMongocryptdIsUnavailableByTest(stdClass $test): void
+    {
+        $clientOptions = isset($test->clientOptions) ? (array) $test->clientOptions : [];
+        $autoEncryptionOptions = isset($clientOptions['autoEncryptOpts']) ? (array) $clientOptions['autoEncryptOpts'] : [];
+        $extraOptions = isset($autoEncryptionOptions['extraOptions']) ? (array) $autoEncryptionOptions['extraOptions'] : [];
+        $mongocryptdURI = $extraOptions['mongocryptdURI'] ?? null;
+
+        $this->skipIfMongocryptdIsUnavailable($mongocryptdURI);
+    }
+
+    protected function skipIfMongocryptdIsUnavailableByAutoEncryptOpts(array $autoEncryptionOpts): void
+    {
+        $mongocryptdURI = $autoEncryptionOpts['extraOptions']['mongocryptdURI'] ?? null;
+
+        $this->skipIfMongocryptdIsUnavailable($mongocryptdURI);
+    }
+
+    protected function skipIfMongocryptdIsUnavailable(?string $mongocryptdURI = null): void
+    {
+        // Assuming that in specific cases with remote mongocryptd developer is responsible for mongocryptd to be available
+        if (! $mongocryptdURI || strpos($mongocryptdURI, 'mongodb://localhost') === 0) {
```

Comment:
Was there a specific reason to add this logic, such as your container configuration? Since the CSFLE spec tests aren't specifying custom URIs, I think it's preferable to just limit the logic to checking whether `mongocryptd` appears within one of the `PATH` directories. That would be sufficient for the needs of PHPLIB's test suite, and also limit the amount of code introduced by this PR.